### PR TITLE
Declarative config: do not create default group if minimum role name is empty

### PR DESCRIPTION
**Pull request by ivan-degtiarenko**
## Description

Empty minimum role name = no default group.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### CODE REVIEW

File: pkg/declarativeconfig/transform/auth_provider.go [hunk 1]
```diff
@@ -144,18 +145,21 @@ func getClaimMappings(claimMappingsConfig []declarativeconfig.ClaimMapping) map[
...
-			AuthProviderId: authProviderID,
-			Key:            "",
-			Value:          "",
-		},
-		RoleName: authProviderConfig.MinimumRoleName,
-	})
+	hasMinimumRoleName := authProviderConfig.MinimumRoleName != ""
+	groups := make([]proto.Message, 0, len(authProviderConfig.Groups)+utils.IfThenElse(hasMinimumRoleName, 1, 0))
+
+	if hasMinimumRoleName {
```
Comment (dhaus67) [inline pkg/declarativeconfig/transform/auth_provider.go:151]: Let's also make sure we adjust the unit test to cover this.

**Review by dhaus67 — APPROVED** [review]
🚢

**Comment by ivan-degtiarenko** [comment]
`ci/prow/ocp-4-12-qa-e2e-tests` is flaky, failed run is unrelated to changes. Merging the PR.

**Pull request merged**